tests/test_project.py

```diff
@@ -348,12 +369,72 @@ def test_context(job):
             assert os.path.realpath(os.getcwd()) == os.path.realpath(job.ws)
 
         project = self.mock_project(A)
-        with _add_cwd_to_environment_pythonpath():
-            with _switch_to_directory(project.root_directory()):
-                starting_dir = os.getcwd()
-                with redirect_stderr(StringIO()):
-                    project.run()
-                assert os.getcwd() == starting_dir
+        with setup_project_subprocess_execution(project):
+            starting_dir = os.getcwd()
+            project.run()
+            assert os.getcwd() == starting_dir
+
+    def test_cmd_operation_argument_as_command(self):
+        class A(FlowProject):
+            pass
+
+        @A.operation(with_job=True, cmd=True)
+        def test_cmd(joba, jobb="test"):
+            return "echo '{joba} {jobb}' > output.txt"
+
+        project = self.mock_project(A)
+        with setup_project_subprocess_execution(project):
+            project.run()
+        for job in project:
+            assert os.path.isfile(job.fn("output.txt"))
+            with open(job.fn("output.txt")) as f:
+                lines = f.readlines()
+                assert f"{job.id} test\n" == lines[0]
```

Comment:
Try to perform `assert` statements _outside_ the context manager if possible. Here, the `lines` have already been read, so we can close the file before comparing to `lines[0]`.
```suggestion
            assert f"{job.id} test\n" == lines[0]
```